File: 이건희/level2/2개 이하로 다른 비트.py
def solution(numbers):
    result = []
    
    for i in numbers:
        if i % 2 == 0:
            result.append(i + 1)
        else:
            num =  format(i, 'b')

            if num.count('0') > 0:
                index = num[::-1].find('0')
                num1 = '1' + '0' * (index)
            else:
                num1 = '1' + '0' * len(num)

            result.append(int(num, 2) + int(num1, 2) // 2)
            
    return result

File: 김찬우/프로그래머스 /Lv.2/2개 이하로 다른 비트.java
import java.util.*;
class Solution {
    public long[] solution(long[] numbers) {
        
        long[] arr = new long[numbers.length];
        StringBuilder sb = new StringBuilder();
        
        for(int i=0; i<numbers.length; i++){
            sb.setLength(0);
            
            
            long a = numbers[i];
          
            if(a%2==0){
                arr[i] = a+1;
                continue;
            }
            
            sb.append(Long.toBinaryString(a));
            
            
            if(sb.indexOf("0") ==-1){
                 sb.insert(0,"0");
            }
            
           
            
            int index = sb.lastIndexOf("01");
            
            String result = sb.substring(0,index) + "10" + sb.substring(index+2);
            
            arr[i] = Long.parseLong(result, 2);
            
            
            
        }
        
        
        


        return arr;
    }
}
